src/background/FinchApi.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  graphql,
+  validate,
+  GraphQLSchema,
+  DocumentNode,
+  print,
+  GraphQLError,
+} from "graphql";
+import gql from "graphql-tag";
+import { applyMiddleware } from "graphql-middleware";
+import { makeExecutableSchema } from "@graphql-tools/schema";
+import { isDocumentNode } from "../utils";
+import {
+  FinchApiOptions,
+  GenericVariables,
+  FinchMessage,
+  FinchMessageKey,
+  FinchContext,
+  FinchMessageSource,
+  FinchContextObj,
+} from "../types";
+import { addExteneralMessageListener, addMessageListener } from "../browser";
+import { NoIntrospection } from "./NoIntrospection";
+
+export class FinchApi {
+  schema: GraphQLSchema;
+  context: FinchContext;
+  onQueryResponse: FinchApiOptions["onQueryResponse"];
+  messageKey?: string;
+  disableIntrospection: boolean;
+  rules: any[];
+  constructor({
+    context,
+    attachMessages,
+    attachExternalMessages,
+    messageKey,
+    onQueryResponse = () => {},
+    disableIntrospection,
+    validationRules = [],
+    ...options
+  }: FinchApiOptions) {
+    this.schema = makeExecutableSchema(options);
+    if (options.middleware) {
```

Comment:
Sorry moving this line made this confusing. This is the only change in this file.